wgsl/index.bs

```diff
@@ -6981,13 +6979,310 @@ See [[#statements]] and [[#function-calls]].
+
+In the table below, `(CF1, S) -> CF2` means "run the analysis on S starting with control flow CF1, apply the required changes to the graph, and name the resulting control flow CF2".
+Similarly, `(CF1, E) -> (CF2, V)` means "run the analysis on expression E, starting with control flow CF1, apply the required changes to the graph, and name the resulting control flow node CF2 and the resulting value node V" (see next section for the analysis of expressions).
+
+We have a similar set of rules for expressions in left-value positions, that we denote by `(CF, E) => (CF, L)`. Instead of computing the node which corresponds to the uniformity of the value, it computes the node which corresponds to the uniformity of the variable we are adressing.
+
+<table class='data'>
+  <caption>Uniformity rules for statements</caption>
+  <thead>
+    <tr><th>Statement<th>New nodes<th>Recursive analyses<th>Resulting control flow node<th>New edges
+  </thead>
+  <tr><td class="nowrap">*s1*; *s2*
+      <td>
+      <td class = "nowrap">(*CF*, *s1*) -> *CF1*<br>
+          (*CF1*, *s2*) -> *CF2*
+      <td>*CF2*
+      <td>
+  <tr><td class="nowrap">if (*e*) *s1* else *s2*
+      <td>*CFinside*, *CFend*
+      <td class="nowrap">(*CF*, *e*) -> (*CF'*, *V*)<br>
+          (*CFinside*, *s1*) -> *CF1*<br>
+          (*CFinside*, *s2*) -> *CF2*
+      <td>*CFend*
+      <td class="nowrap">*CFinside* -> *V*<br>
+          *CFend* -> {*CF1*, *CF2*}
+  <tr><td class="nowrap">loop {*s1* continuing {*s2*}}
+      <td>*CF'*
+      <td class="nowrap">(*CF'*, *s1*) -> *CF1*<br>
+          (*CF1*, *s2*) -> *CF2*
+      <td>*CF'*
+      <td class="nowrap">*CF'* -> {*CF2*, *CF*}
+  <tr><td class="nowrap">loop {*s1*}
+      <td>*CF'*
+      <td class="nowrap">(*CF'*, *s1*) -> *CF1*
+      <td>*CF'*
+      <td class="nowrap">*CF'* -> {*CF1*, *CF*}
+  <tr><td class="nowrap">break;
+      <td>
+      <td>
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">continue;
+      <td>
+      <td>
+      <td>*CF*
+      <td>
+<tr><td class="nowrap">fallthrough;
+      <td>
+      <td>
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">discard;
+      <td>
+      <td>
+      <td>*CF*
+      <td>
+  <tr><td class="nowrap">return;
+      <td>
+      <td>
+      <td>*CF*
+      <td>*CF_return* -> *CF*
+  <tr><td class="nowrap">return *e*;
+      <td>
+      <td class="nowrap">(*CF*, *e*) -> (*CF'*, *V*)
+      <td>*CF'*
+      <td>*CF_return* -> *CF'*<br>
+          *Value_return* -> *V*
+  <tr><td class="nowrap">e1 = e2;
+      <td>
+      <td class="nowrap">(*CF*, *e1*) => (*CF1*, *L1*)<br>
+          (*CF1*, *e2*) -> (*CF2*, *V2*)
+      <td>*CF2*
+      <td>*L1* -> *V2*
+</table>
+
+TODO: double check the rule for loop. I had verified the rules for "for" and "while", but I have some difficulty wrapping my brain around the "continuing" block.
+
+### Uniformity rules for function calls ### {#uniformity-function-calls}
+
+The most complex rule is for function calls:
```

Comment:
nit: maybe remove "complex" from here?

I think that with the suggestion for the rephrasing of the metadata, this algorithm is simplified because the metadata is directly the list of edges that we need to add (plus the CF_after -> CF_i_last)

Should we note that the linking of CF_i between each other is due to the how the execution order is specified?